sources/commands/Cleanup.ts

```diff
@@ -0,0 +1,90 @@
+import {Command}          from 'clipanion';
+import fs                 from 'fs';
+import path               from 'path';
+import semver             from 'semver';
+
+import * as debugUtils    from '../debugUtils';
+import {getInstallFolder} from '../folderUtils';
+import type {Context}     from '../main';
+import type {NodeError}   from '../nodeUtils';
+import {parseSpec}        from '../specUtils';
+
+export class CleanupCommand extends Command<Context> {
+  static paths = [
+    [`cleanup`],
+  ];
+
+  static usage = Command.Usage({
+    description: `Cleans Corepack cache`,
+    details: `
+      When run, this commmand will check what are the versions required by the package.json files it knows of, and remove from the cache all the versions that are in used.
+    `,
+  });
+
+  async execute() {
+    const installFolder = getInstallFolder();
+    const listFile = await fs.promises.open(path.join(installFolder, `packageJsonList.json`), `r+`);
+
+    const previousList = JSON.parse(await listFile.readFile(`utf8`)) as Array<string>;
+    const listFilteredOffOfInvalidManifests = new Set<string>();
+    const inusedSpecs = [];
+
+    for (const pkgPath of previousList) {
+      let pkgContent: string;
+      try {
+        pkgContent = await fs.promises.readFile(pkgPath, `utf8`);
```

Comment:
Shouldn't we just always remove the whole folder? If it's still needed it can be redownloaded, it's not a huge issue (it's already how package managers work).

Reply:
The idea I had was we might want to have an auto-cleaning of the cache. Removing the whole folder is certainly simpler.

Reply:
I agree - start with a simple delete of the whole directory and ship that.

Then we can bikeshed the syntax for partial cleanup in a new PR or continue discussing in Issue #114